treesit-fold.el

```diff
@@ -63,6 +62,7 @@
     (agda-mode              . ,(treesit-fold-parsers-agda))
     (arduino-mode           . ,(treesit-fold-parsers-arduino))
     (asm-mode               . ,(treesit-fold-parsers-asm))
+    (awk-mode               . ,(treesit-fold-parsers-c))
```

Comment:
Should we create a specific parser function for Awk? 🤔

Reply:
as far as i know awk uses some of c syntax, such as `if`, `switch`, `for` and `while`.
so i don't think it should be necessary (for now)

Reply:
I prefer to wrap it like this: 🤔 

```elisp
(defun treesit-fold-parsers-awk ()
  "Rule set for Awk."
  (append (treesit-fold-parsers-c)))
```

And [tree-sitter-awk](https://github.com/Beaglefoot/tree-sitter-awk) doesn't seem to build on top of the tree-sitter-c. Maybe add a comment `;; TODO:`besides it. 🤔

Reply:
> > It can be useful in some cases such as fold large lists, or large conditionals
> 
> My concern is most of the statements are list. Like `(require 'xxx)` is also identified as a list. 🤔

it only would get folded if it's something like:
```elisp
(require 'subr-x
         nil
         t)
```
which i think it would be very rare that someone indent it like that.

> I prefer to wrap it like this: 🤔
> 
> ```emacs-lisp
> (defun treesit-fold-parsers-awk ()
>   "Rule set for Awk."
>   (append (treesit-fold-parsers-c)))
> ```
> 
> And [tree-sitter-awk](https://github.com/Beaglefoot/tree-sitter-awk) doesn't seem to build on top of the tree-sitter-c. Maybe add a comment `;; TODO:`besides it. 🤔

Done